### PR TITLE
CLC-4663 Remove unnecessary parsing from Bearfacts API clients

### PR DESCRIPTION
**Pull request by paulkerschen**
Part two of the unexpectedly expansive https://jira.ets.berkeley.edu/jira/browse/CLC-4663.

### CODE REVIEW

File: app/models/bearfacts/proxy.rb [hunk 1]
```diff
@@ -23,35 +23,19 @@ def request(path, vcr_cassette, params = {})
...
     end
 
     def request_internal(path, vcr_cassette, params = {})
       student_id = lookup_student_id
       if student_id.nil?
-        logger.info "Lookup of student_id for uid #@uid failed, cannot call Bearfacts API path #{path}"
-        return {
-          statusCode: 400
-        }
+        raise Errors::ProxyError.new("Lookup of student_id for uid #@uid failed, cannot call Bearfacts API path #{path}", {noStudentId: true})
```
Comment (johncrossman) [inline app/models/bearfacts/proxy.rb]: Braces needed on @uid for interpolation in error string.

File: app/models/my_academics/college_and_level.rb [hunk 2]
```diff
@@ -4,56 +4,57 @@ class CollegeAndLevel
...
         else
           logger.error("Unknown ugGradFlag '#{ug_grad_flag}' for user #{@uid}")
-          return
-      end
-      general_profile = doc.css("studentGeneralProfile")
-      level = to_text(general_profile.css("corpEducLevel")).titleize
-      nonAPLevel = to_text(general_profile.css("nonAPLevel")).titleize
-      futureTBLevel = to_text(general_profile.css("futureTBLevel")).titleize
+          return {}
+        end
```
Comment (johncrossman) [inline app/models/my_academics/college_and_level.rb]: Do we need to align 'end' with 'standing = ...'?

File: app/models/my_academics/telebears.rb [hunk 3]
```diff
@@ -14,19 +14,19 @@ def merge(data)
...
+      slug = "#{term.downcase}-#{year}"
 
-      auth_release_code = doc.at_css("telebearsAppointment authReleaseCode").text.strip rescue return
+      auth_release_code = feed['telebearsAppointment']['authReleaseCode'].to_text
+      return if auth_release_code.blank?
       adviser_code_required = decode_adviser_code_required(auth_release_code)
-      phases = parse_appointment_phases(doc.css("telebearsAppointment telebearsAppointmentPhase") || [])
-      slug = "#{term.downcase}-#{year}"
+
+      phases = parse_appointment_phases(feed['telebearsAppointment']['telebearsAppointmentPhase'].to_a)
```
Comment (johncrossman) [inline app/models/my_academics/telebears.rb]: Remove parentheses for readability?

**Comment by johncrossman** [comment]
Nice work, @paulkerschen. Looks good. I commented on a couple style issues. But given the extent of changes I'd like one more pair of eyes (@raydavis or @redconfetti) to sanity check the logic.

**Comment by christianv** [comment]
@paulkerschen Thanks for this clean-up!

**Comment by paulkerschen** [comment]
Thanks for the style catches, @johncrossman . I've pushed up the edits but I'm sure that @raydavis will want to have a look at this before it goes in.

**Comment by raydavis** [comment]
188 additions and 301 deletions? What's not to like!?

**Pull request merged**